### PR TITLE
Add vade-sidetree typings

### PR DESCRIPTION
**Pull request by S3bb1**
This PR adds the sidetree typings for typescript

### CODE REVIEW

File: typings/vade_sidetree.d.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+/**
+ * Defines the result of a DID resolution operation.
+ *
+ * @see {@link Resolvable.resolve}
+ * @see {@link https://www.w3.org/TR/did-core/#did-resolution}
+ */
+export interface DIDResolutionResult {
+  '@context'?: 'https://w3id.org/did-resolution/v1' | string | string[];
+  didResolutionMetadata: DIDResolutionMetadata;
```
Comment (vineetpant) [inline typings/vade_sidetree.d.ts]: type DIDResolutionMetadata is missing  ? Getting error 

=> Cannot find name 'DIDResolutionMetadata'.ts(2304)

File: typings/vade_sidetree.d.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  didDocument: DIDDocument;
+  didDocumentMetadata: DidDocumentMetadata;
+}
+
+export interface DidUpdateOptions {
+  type: string;
+  waitForCompletion?: boolean;
+}
+export interface DidUpdatePayload {
+  updateType: UpdateType;
```
Comment (vineetpant) [inline typings/vade_sidetree.d.ts:185]: type UpdateType missing ? getting error => Cannot find name 'UpdateType'.ts(2304)

Comment (S3bb1) [reply typings/vade_sidetree.d.ts:185]: included now :)

File: typings/vade_sidetree.d.ts (hunk 1 above)
Comment (S3bb1) [reply typings/vade_sidetree.d.ts]: removed, because its unused

File: typings/vade_sidetree.d.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+  controller?: string;
+}
+
+export enum PublicKeyPurpose {
+  Authentication = 'authentication',
+  AssertionMethod = 'assertionMethod',
+  CapabilityInvocation = 'capabilityInvocation',
+  CapabilityDelegation = 'capabilityDelegation',
+  KeyAgreement = 'keyAgreement',
+}
```
Comment (vineetpant) [inline typings/vade_sidetree.d.ts]: PublicKeyPurpose enum defined twice, it is also defined @ line #23
```suggestion
```

**Review by vineetpant — APPROVED** [review]

**Review by wulfraem — APPROVED** [review]

**Pull request merged**